Explorer/Assets/DCL/WebRequests/SafeMainThreadWebRequestController.cs

```diff
@@ -0,0 +1,21 @@
+using Cysharp.Threading.Tasks;
+using Utility.Multithreading;
+
+namespace DCL.WebRequests
+{
+    public class SafeMainThreadWebRequestController : IWebRequestController
+    {
+        private readonly IWebRequestController origin;
+
+        public SafeMainThreadWebRequestController(IWebRequestController origin)
+        {
+            this.origin = origin;
+        }
+
+        public async UniTask<TResult?> SendAsync<TWebRequest, TWebRequestArgs, TWebRequestOp, TResult>(RequestEnvelope<TWebRequest, TWebRequestArgs> envelope, TWebRequestOp op) where TWebRequest: struct, ITypedWebRequest where TWebRequestArgs: struct where TWebRequestOp: IWebRequestOp<TWebRequest, TResult>
```

Comment:
I wonder why we need a new class if the underlying core can't work on the background thread anyway. Thus (as I believe you wanted to secure `UntiyWebRequest` creation), switching to the main thread is not detachable. 
So you can simply add `await UniTask.SwitchToMainThread();` to the original method

Reply:
I got an error with PlayerPrefs on trying to access it from background thread. So somewhere in the further flow it switches to a background thread (maybe on parsing). And I decided to secure it. In addition with this approach the caller side can don't care about it's calling from main or background threads

Reply:
Can you present the error because parsing is not the responsibility of `WebRequestController`? 
Still don't you think that you can just move switching to the main thread logic to the `WebRequestController` itself as it's already known that WebRequests can't be created from the background thread and therefore no wrapper is needed?

Reply:
We shouldn't move switching logic to `WebRequestController` because we have calls to Unity Api (PlayerPrefs) on the layer above and it could lead to "can only be called from the main thread"

What do you think about it?

Reply:
On the layer above? Can you give a link? This decorator operates on the same later as the original `WebRequestController`. There is no layer between the decorator and the controller. How can it help layers above?

Reply:
One more thing: the name of `NewScopeWithReturnOnOriginalThreadAsync` is misleading: it does not return to the original thread but to the thread pool

Reply:
Talking about layers I meant decorator. So okay, we could move the switch to main thread logic directly to `WebRequestController` that may lead to "can only be called from the main thread" due PlayerPrefs. I don't see a point here to make a danger point and make developers to memorize it when it's possible to safely cover this place

Reply:
So we found a common ground here in the private conversation: `SafeMainThread` decorator will go away and `Delay` will provide its own thread-safety for its own needs